scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -300,6 +301,25 @@
          (catch Exception e
            (log/error e "Mesos scheduler status update error")))))
 
+(defn write-sandbox-url-to-datomic
+  "Takes a sandbox file server URL from the compute cluster and saves it to datomic."
+  [conn task-id sandbox-url]
+  (try (let [db (db conn)
+             _ (when-not task-id
+                 (throw (ex-info "task-id is nil. Something unexpected has happened."
```

Comment:
Why not lift this outside the `try` block and just log the error rather than `throw`ing?